crawl.py

```diff
@@ -1,47 +1,75 @@
 #!/usr/bin/env python3
 
 import json
-from json.decoder import JSONDecodeError
 import requests
+from json.decoder import JSONDecodeError
 import csv
 import argparse
 import signal
 import sys
+import itertools
 import traceback
 import os
+from concurrent.futures import ThreadPoolExecutor
+from tenacity import retry, stop_after_attempt, wait_exponential, retry_if_exception_type
+import threading
+
+# Initialize global resources
+output_file = None
+executor = None
 
+# Argument parsing
 parser = argparse.ArgumentParser(description='Make requests to Cookiemonster API')
 parser.add_argument('-i', '--input-file', dest='input', required=True, help='Path to the input file')
 parser.add_argument('-o', '--output-file', dest='output', required=True, help='Path to the output file')
-parser.add_argument('-s', '--skip', dest='skip', type=int, default=0, help='Rows to skip from input CSV file (default: 0)')
-parser.add_argument('-c', '--skip-vpn-check', dest='skip_vpn', default=False, help='Skip check for company VPN before starting crawl (default: False)')
+parser.add_argument('-s', '--skip-lines', dest='skip_lines', type=int, default=0, help='Rows to skip from input CSV file (default: 0)')
+parser.add_argument('-v', '--skip-vpn-check', action='store_true', dest='skip_vpn', help='Skip VPN check (default: required)')
+parser.add_argument('-p', '--parallel', dest='parallel', action='store_true', help='Enable parallel crawling (default: sequential)')
```

Comment:
`(default: required)` is somewhat confusing?

FWIW I think the default behavior is already clear enough without any callout here, since it's a binary toggle flag and doesn't take a value